governance/fma-isthmus-l2-pectra.md

```diff
@@ -0,0 +1,164 @@
+- **Description:** If implemented in the FP program, BLS precompiles could fail to execute. This could cause certain blocks to be unprovable.
+- **Risk Assessment:** High impact, low likelihood
+- **Mitigations:**
+	1. Fully mitigated by using an accelerated precompile that calls out to L1 instead of calculating inside the program. This means it should 
+  have the same performance characteristics as existing crypto precompiles like `ecrecover`.
+- **Detection:** op-program trace runner failure (possibly triggering an alert)
+- **Recovery Path(s)**: If this occurred, which it can't with the mitigations, we'd have to make op-program efficient enough to execute BLSprecompiles in sufficient time.
+
+### FM6: Increased call data cost affects network upgrade transactions
+
+- **Description:** Increased calldata cost could mean that network upgrade transactions fail.
+- **Risk Assessment:** High impact, low likelihood
+- **Mitigations:**
+	1. Tested that network upgrade transactions don't fail.
+- **Detection:** op-program trace runner failure (possibly triggering an alert)
+- **Recovery Path(s)**: If this occurred, which it can't with the mitigations, we'd have to make op-program efficient enough to execute BLSprecompiles in sufficient time.
+
+### FM7: Early fork if batches containing EIP-7702 transactions could be posted before Pectra
+
+- **Description:** If batches containing EIP-7702 transactions are posted and accepted early, the batcher could fork any nodes that upgrade early.
+- **Risk Assessment:** High impact, low likelihood
+- **Mitigations:**
+	1. End-to-end test
+- **Detection:** Upgraded clients would fork
+- **Recovery Path(s)**: Emergency patch
+
+
+### Generic items we need to take into account:
+
+<!-- See [generic hardfork failure modes](./fma-generic-hardfork.md) and [generic smart contract failure modes](./fma-generic-contracts.md).
+Incorporate any applicable failure modes with FMA-specific mitigations and detections directly into this document. -->
+
+- [x] Check this box to confirm that these items have been considered and updated if necessary.
+
+## Action Items
+
+Below is what needs to be done before launch to reduce the chances of the above failure modes occurring, and to ensure they can be detected and recovered from:
+
+- [ ] Resolve all comments on this document and incorporate them into the document itself (Assignee: document author)
+- [ ] FM6: double check that call data cost doesn't break upgrade transactions
+
+## Audit Requirements
+
+_Given the failure modes and action items, will this project require an audit? See [OP Labs Audit Framework: When to get external security review and how to prepare for it](https://gov.optimism.io/t/op-labs-audit-framework-when-to-get-external-security-review-and-how-to-prepare-for-it/6864) for a reference decision making framework. Please explain your reasoning._
+
+**No.** All of the code is well-tested and has very low likelihood for any failure modes with reasonably high impact.
```

Comment:
We should explain the rationale in the context of the audit framework linked above

Reply:
Added to the end with an explanation on why we should rely on real-world testing rather than auditing.